src/components/who-is-online/index.ts

```diff
@@ -0,0 +1,232 @@
+import { MeetingColorsHex } from '../../common/types/meeting-colors.types';
+import { Logger } from '../../common/utils';
+import { WhoIsOnline as WhoIsOnlineElement } from '../../web-components';
+import { BaseComponent } from '../base';
+import { ComponentNames } from '../types';
+
+import { WhoIsOnlinePosition, Position, Participant } from './types';
+import { AblyParticipant } from '../../services/realtime/ably/types';
+
+export class WhoIsOnline extends BaseComponent {
+  public name: ComponentNames;
+  protected logger: Logger;
+  private element: WhoIsOnlineElement;
+  private position: WhoIsOnlinePosition;
+  private participants: Participant[] = [];
+
+  constructor(position?: WhoIsOnlinePosition) {
+    super();
+    this.name = ComponentNames.WHO_IS_ONLINE;
+    this.position = position ?? Position.TOP_RIGHT;
+    this.logger = new Logger('@superviz/sdk/who-is-online-component');
+  }
+
+  /**
+   * @function start
+   * @description Initializes the Who Is Online component
+   * @returns {void}
+   */
+  protected start(): void {
+    this.subscribeToRealtimeEvents();
+    this.positionWhoIsOnline();
+
+    this.participants = Object.values(this.realtime.getParticipants).map((participant) => {
+      const { slotIndex } = participant.data;
+      const color = MeetingColorsHex[slotIndex];
+
+      return {
+        ...participant.data,
+        participantId: participant.id,
+        color,
+        slotIndex,
+      };
+    });
+  }
+
+  /**
+   * @function destroy
+   * @description Destroys the Who Is Online component
+   * @returns {void}
+   */
+  protected destroy(): void {
+    this.unsubscribeToRealtimeEvents();
+    this.element.remove();
+    this.element = null;
+    this.participants = null;
+  }
+
+  /**
+   * @function subscribeToRealtimeEvents
+   * @description Subscribes to realtime events
+   * @returns {void}
+   */
+  private subscribeToRealtimeEvents(): void {
+    this.realtime.presenceMouseParticipantJoinedObserver.subscribe(this.onParticipantJoined);
+    this.realtime.presenceMouseParticipantLeaveObserver.subscribe(this.onParticipantLeave);
+    this.realtime.participantsObserver.subscribe(this.onParticipantListUpdate);
+  }
+
+  /**
+   * @function subscribeToRealtimeEvents
+   * @description Unsubscribes to realtime events
+   * @returns {void}
+   */
+  private unsubscribeToRealtimeEvents(): void {
+    this.realtime.presenceMouseParticipantJoinedObserver.unsubscribe(this.onParticipantJoined);
+    this.realtime.presenceMouseParticipantLeaveObserver.unsubscribe(this.onParticipantLeave);
+    this.realtime.participantsObserver.unsubscribe(this.onParticipantListUpdate);
+  }
+
+  private compareParticipants() {
+    const realtimeParticipants = this.realtime.getParticipants;
+
+    for (let i = 0; i < this.participants.length; i++) {
+      if (!realtimeParticipants[this.participants[i].id]) return false;
+    }
+
+    return true;
+  }
```

Comment:
`const allParticipantsOnline = this.participants.every(participant => 
  !!realtimeParticipants[participant.id]
);

return allParticipantsOnline;`

In this case, you can use "some" or "every" depending on what you want to retrieve from the array. You can test to see if it works. Remember that "some" returns true if it finds a match, while "every" checks all elements.